llvm/include/llvm/SYCLLowerIR/GlobalOffset.h

```diff
@@ -5,21 +5,117 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-//
-// This pass operates on SYCL kernels being compiled to CUDA. It looks for uses
-// of the `llvm.nvvm.implicit.offset` intrinsic and replaces it with a offset
-// parameter which will be threaded through from the kernel entry point.
-//
-//===----------------------------------------------------------------------===//
 
 #ifndef LLVM_SYCL_GLOBALOFFSET_H
 #define LLVM_SYCL_GLOBALOFFSET_H
 
-#include "llvm/Pass.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/SYCLLowerIR/TargetHelpers.h"
 
 namespace llvm {
 
-ModulePass *createGlobalOffsetPass();
+class ModulePass;
+class PassRegistry;
+
+/// This pass operates on SYCL kernels. It looks for uses of the
```

Comment:
Can this comment be: This pass operates on SYCL kernels that target AMDGPU or NVVM.